tree/ntuple/v7/test/ntuple_descriptor.cxx

```diff
@@ -480,3 +482,57 @@ TEST(RNTupleDescriptor, Clone)
    auto clone = desc.Clone();
    EXPECT_EQ(desc, *clone);
 }
+
+TEST(RNTupleDescriptor, BuildStreamerInfos)
+{
+   auto fnBuildStreamerInfosOf = [](const RFieldBase &field) -> RNTupleSerializer::StreamerInfoMap_t {
+      RNTupleDescriptorBuilder descBuilder;
+      descBuilder.SetNTuple("test", "");
+      descBuilder.AddField(
+         RFieldDescriptorBuilder().FieldId(0).Structure(ENTupleStructure::kRecord).MakeDescriptor().Unwrap());
+      auto fieldBuilder = RFieldDescriptorBuilder::FromField(field);
+      descBuilder.AddField(fieldBuilder.FieldId(1).MakeDescriptor().Unwrap());
+      descBuilder.AddFieldLink(0, 1);
+      int i = 2;
+      // In this test, we only support leaf fields or fields with only direct sub fields
+      for (const auto &subField : field.GetSubFields()) {
+         fieldBuilder = RFieldDescriptorBuilder::FromField(*subField);
+         descBuilder.AddField(fieldBuilder.FieldId(i).MakeDescriptor().Unwrap());
+         descBuilder.AddFieldLink(1, i);
+         i++;
+      }
+      return descBuilder.BuildStreamerInfos();
+   };
+
+   RNTupleSerializer::StreamerInfoMap_t streamerInfoMap;
+
+   streamerInfoMap = fnBuildStreamerInfosOf(*RFieldBase::Create("f", "float").Unwrap());
+   EXPECT_TRUE(streamerInfoMap.empty());
+
+   streamerInfoMap = fnBuildStreamerInfosOf(*RFieldBase::Create("f", "std::vector<float>").Unwrap());
+   EXPECT_TRUE(streamerInfoMap.empty());
+
+   streamerInfoMap = fnBuildStreamerInfosOf(*RFieldBase::Create("f", "std::pair<float, float>").Unwrap());
+   EXPECT_TRUE(streamerInfoMap.empty());
+
+   streamerInfoMap = fnBuildStreamerInfosOf(ROOT::Experimental::RRecordField("f", {}));
+   EXPECT_TRUE(streamerInfoMap.empty());
+
+   streamerInfoMap = fnBuildStreamerInfosOf(*RFieldBase::Create("f", "CustomStruct").Unwrap());
+   EXPECT_EQ(1u, streamerInfoMap.size());
+   EXPECT_STREQ("CustomStruct", streamerInfoMap.begin()->second->GetName());
+
+   streamerInfoMap = fnBuildStreamerInfosOf(*RFieldBase::Create("f", "std::vector<CustomStruct>").Unwrap());
```

Comment:
maybe test `std::pair<CustomStruct, DerivedA>` (to test deduplication of streamer infos) and `std::[unordered_]map<int, CustomStruct>` as well?